src/NzbDrone.Common/Extensions/IpAddressExtensions.cs

```diff
@@ -50,7 +59,10 @@ private static bool IsLocalIPv4(byte[] ipv4Bytes)
             // Class C private range: 192.168.0.0 – 192.168.255.255 (192.168.0.0/16)
             bool IsClassC() => ipv4Bytes[0] == 192 && ipv4Bytes[1] == 168;
 
-            return IsLinkLocal() || IsClassA() || IsClassC() || IsClassB();
+            // CGNAT range: 100.64.0.0 - 100.127.255.255 (100.64.0.0/10)
+            bool IsCGNAT() => ipv4Bytes[0] == 100 && ipv4Bytes[1] >= 64 && ipv4Bytes[1] <= 127;
+
+            return IsLinkLocal() || IsClassA() || IsClassC() || IsClassB() || (TrustCGNAT && IsCGNAT());
```

Comment:
This is no longer checking if it's a local IP and should be done elsewhere, probably where `IsLocalAddress` is called from.